src/rpc/core_rpc_server.cpp

```diff
@@ -2134,6 +2128,13 @@ namespace cryptonote
           d.cached = true;
         }
 
+        if (req.cumulative)
+        {
+          distribution[0] += base;
+          for (size_t n = 1; n < distribution.size(); ++n)
+            distribution[n] += distribution[n-1];
+        }
+
```

Comment:
Isn't it necessary to also do the conversion when returning the cached data in the above `if (d.cached && amount == 0 && d.cached_from == req.from_height && d.cached_to == req.to_height) {...}`?

Reply:
Yes

Reply:
3900